### PR TITLE
http: Use libhtp-rs.

### PR DESCRIPTION
**Pull request by catenacyber**
Link to [redmine](https://redmine.openinfosecfoundation.org/issues/2696) ticket: 2696

Describe changes:
- Use libhtp-rs.

Clean rebase from #11468
New in htp rs:
- Fix latest oss-fuzz quadratic complexity in htp rs
- rust: fix warnings
- time: upgrade to version used by Suricata
- headers: put a configurable limit on their numbers (following C commit)

Draft to see CI feedback with latest libhtp-rs changes, and not have too many conflicts

TODOs :
- look into the remaining oss-fuzz differential reports for libhtp-rs
- look into QA results

### CODE REVIEW

**Comment by suricata-qa** [comment]
WARNING:
 ERROR: SEGMENTATION FAULT in build_asan QA test
 | field | baseline | test | % |
|--|--|--|--| 
| | build_asan | | |
 

Pipeline 22998

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_single_alerts_cmp.

ERROR: QA failed on SURI_TLPW2_autofp_alerts_cmp.

ERROR: QA failed on SURI_TLPW1_files_sha256.

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_single_stats_chk | | |
| .uptime | 782 | 816 | 104.35% |
| | SURI_TLPW2_autofp_stats_chk | | |
| .flow.end.tcp_state.established | 8 | 9 | 112.5% |
| | SURI_TLPW1_stats_chk | | |
| .app_layer.error.http.parser | 47 | 2 | 4.26% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.error.http.parser | 700 | 369 | 52.71% | 

Pipeline 22999

**Comment by suricata-qa** [comment]
Information:
 

ERROR: QA failed on SURI_TLPW2_single_alerts_cmp.

ERROR: QA failed on SURI_TLPW2_autofp_alerts_cmp.

ERROR: QA failed on SURI_TLPW1_files_sha256.

ERROR: QA failed on SURI_TLPR1_alerts_cmp.
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_single_stats_chk | | |
| .uptime | 782 | 815 | 104.22% |
| | SURI_TLPW2_autofp_stats_chk | | |
| .flow.end.tcp_state.established | 8 | 10 | 125.0% |
| | SURI_TLPW1_stats_chk | | |
| .app_layer.error.http.parser | 47 | 2 | 4.26% |
| | SURI_TLPR1_stats_chk | | |
| .app_layer.error.http.parser | 700 | 369 | 52.71% | 

Pipeline 23039

File: rust/htp/.gitignore [hunk 1]
```diff
@@ -0,0 +1,5 @@
+!Cargo.toml
+/Cargo.lock
+/target
+/vendor
+/gen
```
Comment (jasonish) [inline rust/htp/.gitignore:5]: Being part of the work space, these files are never created.

File: rust/htp/README.md [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+ 4. Performance; The performance must be adequate for the desired tasks.
+    Completeness and security are often detrimental to performance. Our
+    idea of handling the conflicting requirements is to put the library
+    user in control, allowing him to choose the most desired library
+    characteristic.
+
+ | IMPORTANT   LIBHTP IS NOT YET CONSIDERED STABLE. USE AT YOUR OWN RISK. DO NOT  
+ |             USE IN PRODUCTION. WORK IS CURRENTLY UNDER WAY TO ENSURE THAT  
+ |             LIBHTP IS SECURE AND THAT IT PERFORMS WELL.
```
Comment (jasonish) [inline rust/htp/README.md:33]: :)

Comment (catenacyber) [reply rust/htp/README.md:33]: well spotted

File: rust/htp/.gitignore (hunk 1 above)
Comment (jasonish) [reply rust/htp/.gitignore:5]: You probably need `!Cargo.toml` as its ignored in the `.gitignore` in the parent directory.

**Comment by catenacyber** [comment]
Next in https://github.com/OISF/suricata/pull/12050

**Pull request closed without merging**